bpf/lib/nodeport.h

```diff
@@ -238,6 +238,34 @@ static __always_inline int dsr_set_ext6(struct __ctx_buff *ctx,
 		return DROP_INVALID;
 	return 0;
 }
+#elif DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+static __always_inline int dsr_encap_geneve_opt6(struct __ctx_buff *ctx,
+						 struct ipv6hdr *ip6,
+						 const union v6addr *svc_addr,
+						 __be16 svc_port,
+						 int *ifindex)
+{
+	struct remote_endpoint_info *info;
+	union v6addr *dst;
+	struct geneve_opt6 gopt;
+
+	dst = (union v6addr *)&ip6->daddr;
+	info = ipcache_lookup6(&IPCACHE_MAP, dst, V6_CACHE_KEY_LEN, 0);
+	if (info && info->tunnel_endpoint != 0) {
+		set_geneve_opt6(svc_port, svc_addr, &gopt);
+
+		return  __encap_with_nodeid_opt6(ctx, info->tunnel_endpoint,
+						 WORLD_ID,
+						 info->sec_label,
+						 NOT_VTEP_DST,
+						 &gopt,
+						 (enum trace_reason)CT_NEW,
+						 TRACE_PAYLOAD_LEN,
+						 ifindex);
+	}
+
+	return CTX_ACT_DROP;
```

Comment:
We shouldn't return CTX_ACT_DROP as error condition from our own code. Can you make this a proper DROP reason?

Reply:
Sure, will do